homeassistant/components/androidtv_remote/media_player.py

```diff
@@ -0,0 +1,163 @@
+"""Media player support for Android TV Remote."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from androidtvremote2 import AndroidTVRemote
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, KNOWN_APPS
+from .entity import AndroidTVRemoteBaseEntity
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Android TV media player entity based on a config entry."""
+    api: AndroidTVRemote = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AndroidTVRemoteMediaPlayerEntity(api, config_entry)])
+
+
+class AndroidTVRemoteMediaPlayerEntity(AndroidTVRemoteBaseEntity, MediaPlayerEntity):
+    """Android TV Remote Media Player Entity."""
+
+    _attr_assumed_state = True
+    _attr_device_class = MediaPlayerDeviceClass.TV
+    _attr_supported_features = (
+        MediaPlayerEntityFeature.PAUSE
+        | MediaPlayerEntityFeature.VOLUME_STEP
+        | MediaPlayerEntityFeature.VOLUME_MUTE
+        | MediaPlayerEntityFeature.VOLUME_SET
+        | MediaPlayerEntityFeature.PREVIOUS_TRACK
+        | MediaPlayerEntityFeature.NEXT_TRACK
+        | MediaPlayerEntityFeature.TURN_ON
+        | MediaPlayerEntityFeature.TURN_OFF
+        | MediaPlayerEntityFeature.PLAY
+        | MediaPlayerEntityFeature.STOP
+        | MediaPlayerEntityFeature.PLAY_MEDIA
+    )
+
+    def __init__(self, api: AndroidTVRemote, config_entry: ConfigEntry) -> None:
+        """Initialize the entity."""
+        super().__init__(api, config_entry)
+
+        self._update_current_app(api.current_app)
+        self._update_volume_info(api.volume_info)
+
+        @callback
+        def current_app_updated(current_app: str) -> None:
+            self._update_current_app(current_app)
+            self.async_write_ha_state()
+
+        @callback
+        def volume_info_updated(volume_info: dict[str, str | bool]) -> None:
+            self._update_volume_info(volume_info)
+            self.async_write_ha_state()
+
+        api.add_current_app_updated_callback(current_app_updated)
+        api.add_volume_info_updated_callback(volume_info_updated)
+
+    def _update_current_app(self, current_app: str) -> None:
+        """Update current app info."""
+        self._attr_app_id = current_app
+        self._attr_app_name = KNOWN_APPS.get(current_app, current_app)
+
+    def _update_volume_info(self, volume_info: dict[str, str | bool]) -> None:
+        """Update volume info."""
+        self._attr_volume_level = int(volume_info["level"]) / 100
```

Comment:
The API returns volume_info["max"]. Why not use that? For my devices I think the max is 10 or 0 because of soundbar, more on that below.
Also if I remember right often the level and max are missing and I think are set to 0. Can you run demo.py of the underlying library and watch the volume logging when the device turns off? I think that's an instance they are missing. Another instance might be if you have connected a soundbar and you have setup the volume to control the soundbar via HDMI CEC. You have VOLUME_STEP support but not VOLUME_SET. Not sure if you can detect that and not include VOLUME_SET in the supported features. Maybe if max is 0 remove the VOLUME_STEP and if is non 0 add it?

Reply:
I added checks and make floating `VOLUME_SET` in supported features.